### ISSUE TITLE
Baseline or finetuned model on UCF101?

### ISSUE THREAD

**Issue by Tord-Zhang** [wandering007/c3d_torch#1]
Is this a pre-trained model on Sports-1m or a finetuned model on UCF101

**Comment by wandering007** [comment]
@mangdian the c3d_v1.0 torch model is pre-trained on Sports-1m, converted from https://www.dropbox.com/s/vr8ckp0pxgbldhs/conv3d_deepnetA_sport1m_iter_1900000?dl=0

**Comment by Tord-Zhang** [comment]
@wandering007  c3d_v1.1 is a finetuned model on UCF101?It is based on resnet,so can it be used to extract features?

**Comment by wandering007** [comment]
@mangdian I haven't converted c3d_v1.1 model. I think c3d_v1.1 is pre-trained on sports-1m and you can easily fine-tune it on ucf101. Of course it can be used to extract features.

**Comment by Tord-Zhang** [comment]
all right.Thank you.you may close this issue.

**Comment by gss-ucas** [comment]
Hi, @wandering007 
Woderful job! But can you release the code that transfers caffe model to torch model. I need C3D caffe model fine-tuned on UCF101.
Thank you very much!!

**Comment by shuangshuangguo** [comment]
Hi, @wandering007
Woderful job! But can you release the code that transfers caffe model to torch model. I need C3D torch model fine-tuned on UCF101.
Thank you very much!!

**Comment by wandering007** [comment]
@gss-ucas Thanks for your attention.   
The transferring job was dirty work and the code may be lost.  
But I can give you some guidelines.   
1. I extracted the weights of C3D caffe model using the code [here](https://github.com/wandering007/C3D/blob/master/tools/get_weights.cpp).   
2. Then convert the binary files to numpy files using python and read numpy files in torch using the [npy4th package](https://github.com/htwaijry/npy4th) (If I don't remember it wrongly).  Or you simply read binary files in torch.  
3. After you got all the torch tensors of weights, you can use the `c3d_v1.0.lua` to create a c3d model and copy the corresponding weight tensors to the model. Finally, save the model.  

Attention: you need to do some dimension transformations on the weight tensor at first conv layer since the conventional input in torch is not BGR-style but RGB-style.   
In fact, you can just use my released torch model file and fine-tune it on UCF-101, the result seems slightly better that the original fine-tuning result in Caffe.  
Good luck!

**Comment by shuangshuangguo** [comment]
@wandering007 Thank you very much!!! I will try as what you have said.

**Comment by shuangshuangguo** [comment]
@wandering007 By the way, I have trained c3d torch from scratch on UCF101 dataset, but when I use softmax layer to predict, I have only 45% clip accuracy, do you have this problem?

**Comment by wandering007** [comment]
@gss-ucas I didn't train c3d from scratch on UCF101. However, 45% is a little bit low, you may need to double-check your code.

**Comment by shuangshuangguo** [comment]
@wandering007 Thanks!
Then, can you release your fine-tuned model on UCF101? _^_

**Comment by wandering007** [comment]
@gss-ucas I think that you'd better try it yourself. Releasing it seems helpless for learning :-)

**Comment by shuangshuangguo** [comment]
@wandering007 well, thank you very much! I don't know how to use your get_weights.cpp code. 
I'm trying to add get_weights.cpp to C3D_v1.1/tools and then compile C3D, I got error always. I'm new to caffe, Could you give me some advice?
Now, I'm trying to use caffe in torch.

**Comment by wandering007** [comment]
@gss-ucas the `get_weights.cpp` file is used for C3D_v1.0. For C3D_v1.1, it is much simpler to extract weights via pycaffe, maybe like [this one](https://github.com/nilboy/extract-caffe-params).

**Comment by shuangshuangguo** [comment]
@wandering007 Thank you very much!!! I have fixed it!!!  _^_  you are so kind_^_
by the way, do you know how to download sports-1m dataset???   Could you give me the url???
I will appreciate it very much!

**Comment by wandering007** [comment]
@gss-ucas  Follow this: https://github.com/gtoderici/sports-1m-dataset/

**Comment by shuangshuangguo** [comment]
@wandering007 Thank you! Best wishes!!
